docs/PyCDE.md

```diff
@@ -2,12 +2,35 @@
 
 PyCDE stands for Python circuit design entry. It is an experimental, opinionated, Python-based fronted for CIRCT's Python bindings. The goal is to make the definition of hardware modules using the bindings simple.
 
-# Installation
-
-## via Pip
+# To Install PyCDE only via pip
 
 PyCDE is now being released on PyPI: https://pypi.org/project/pycde/
 
+# For CIRCT-based Developers
+## Cloning the repo
+
+If you havent already, you need to clone the CIRCT repo. Unless you already
+have contributor permissions to the LLVM project, the easiest way to develop
+(with the ability to create and push branches) is to fork the repo in your
+GitHub account. You can then clone your fork. The clone command should look
+like this:
+
+```
+$ git clone git@github.com:<your_github_username>/circt.git <optional_repo_name>
+```
+
+After cloning, navigate to your repo root (circt is the default) and use the
+following to pull down LLVM:
+
+```
+$ git submodule update --init
+```
+
+If you don't envision needing that ability, you can clone the main repo
```

Comment:
I suggest moving this para up to just below the git clone so it's clear to what you are referring.

Reply:
Moved up to just below the git clone cmd.